src/server/graphql/mutations/login.js

```diff
@@ -0,0 +1,85 @@
+import {GraphQLNonNull, GraphQLString} from 'graphql';
+import {verify} from 'jsonwebtoken';
+import getRethink from 'server/database/rethinkDriver';
+import sendEmail from 'server/email/sendEmail';
+import LoginPayload from 'server/graphql/types/LoginPayload';
+import {
+  auth0AuthenticationClient as auth0Client,
+  clientId as auth0ClientId,
+  clientSecret as auth0ClientSecret
+} from 'server/utils/auth0Helpers';
+import {getUserId} from 'server/utils/authorization';
+import segmentIo from 'server/utils/segmentIo';
+
+const login = {
```

Comment:
> This just verifies a JWT, and maybe creates an account.

Isn't that a perfect description of what logging is does? :wink:

It was called something like updateUserWithToken, but I could never remember that name. This mutation gets called when you log in, it logs you in, so i figure hey, let's call it log in. No strong opinion on keeping the name.

Reply:
I guess that the "creating an account" part doesn't feel like logging in, but I'm nitpicking here!

Reply:
yeah, ever since oauth the line between sign up & log in has been really blurred! How many places say "sign up or log in with google". Well, technically, i wanna sign up with you, but through google. On the backend, that don't matter though, because google just gives us a token, they don't tell us if they're a first timer or not, so we don't even know if they're new!

Reply:
We know if they're new based on whether they're in our `User` table or not.

Reply:
the server knows, but the client doesn't.
if we have a login & signup mutation, which one will the client call?